### PR TITLE
fix: replace double-slashes in `join` calls

### LINKED ISSUE #16: Duplicate slashes with `join`
Repro (pathe: 0.3.0)

```js
require('pathe').join('/', 'foo.js') // "//foo.js"

```

### PR DESCRIPTION
**Pull request by danielroe**
The tests added are all default Node behaviour.

resolves #16

### CODE REVIEW

File: src/path.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ export const join: typeof path.join = function (...args) {
     return '.'
   }
 
-  return normalize(joined)
+  return normalize(joined.replace(/\/\/+/g, '/'))
```
[the remark this answers is not in the document]
Comment (danielroe) [reply src/path.ts:73]: What do you mean by `root cause with our 0.3x migration`? I might be missing some context here.

[the remark this answers is not in the document]
Comment (pi0) [reply src/path.ts:73]: In pathe 0.2.x, the Node.js implementation didn't have this issue (and I guess Node.js implementation is not doing `//`(g) => `/` replacement either). So there might be something we forgot and if fixed, we probably don't need a regex to patch the result.

[the remark this answers is not in the document]
Comment (danielroe) [reply src/path.ts:73]: Yes, we did have this issue: https://stackblitz.com/edit/node-qqdovk?file=index.mjs.

[the remark this answers is not in the document]
Comment (pi0) [reply src/path.ts:73]: mkdist generating duplicate `.d.ts` entries failing override test (https://github.com/unjs/mkdist/commit/659f7e97b0fdc4d0187deade839705597287a895)

[the remark this answers is not in the document]
Comment (danielroe) [reply src/path.ts:73]: the issue in mkdist was caused by #18

**Pull request merged**